components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/revoke/OAuthRevocationEndpoint.java

```diff
@@ -28,271 +27,330 @@
         }
-    }
 
-    private Response handleClientFailure(String callback)
-            throws OAuthSystemException {
-        if (callback == null || "".equals(callback)) {
+        if (isBlank(callback)) {
             OAuthResponse response = OAuthASResponse.errorResponse(HttpServletResponse.SC_BAD_REQUEST)
                     .setError(OAuth2ErrorCodes.INVALID_REQUEST)
                     .setErrorDescription("Invalid revocation request").buildJSONMessage();
             return Response.status(response.getResponseStatus())
-                    .header("Content-Type", "text/html")
+                    .header(HttpHeaders.CONTENT_TYPE, TEXT_HTML)
                     .entity(response.getBody()).build();
         } else {
             OAuthResponse response = OAuthASResponse.errorResponse(HttpServletResponse.SC_BAD_REQUEST)
                     .setError(OAuth2ErrorCodes.INVALID_REQUEST).buildJSONMessage();
             return Response.status(response.getResponseStatus())
-                    .header("Content-Type", "application/javascript")
+                    .header(HttpHeaders.CONTENT_TYPE, APPLICATION_JAVASCRIPT)
                     .entity(callback + "(" + response.getBody() + ");").build();
         }
     }
 
     private Response handleClientFailure(String callback, OAuthRevocationResponseDTO dto)
             throws OAuthSystemException {
-        if (callback == null || "".equals(callback)) {
+
+        if (isBlank(callback)) {
             OAuthResponse response = OAuthASResponse.errorResponse(HttpServletResponse.SC_BAD_REQUEST)
                     .setError(dto.getErrorCode())
                     .setErrorDescription(dto.getErrorMsg()).buildJSONMessage();
             return Response.status(response.getResponseStatus())
-                    .header("Content-Type", "text/html")
+                    .header(HttpHeaders.CONTENT_TYPE, TEXT_HTML)
                     .entity(response.getBody()).build();
         } else {
             OAuthResponse response = OAuthASResponse.errorResponse(HttpServletResponse.SC_BAD_REQUEST)
                     .setError(dto.getErrorCode()).buildJSONMessage();
             return Response.status(response.getResponseStatus())
-                    .header("Content-Type", "application/javascript")
+                    .header(HttpHeaders.CONTENT_TYPE, APPLICATION_JAVASCRIPT)
                     .entity(callback + "(" + response.getBody() + ");").build();
         }
     }
 
-    private OAuthRevocationResponseDTO revokeTokens(OAuthRevocationRequestDTO oauthRequest)
-            throws OAuthClientException {
+    private OAuthRevocationResponseDTO revokeTokens(OAuthRevocationRequestDTO oauthRequest) {
+        OAuthRevocationRequestDTO revokeReqDTO = getoAuthRevocationRequestDTO(oauthRequest);
+        return getOAuth2Service().revokeTokenByOAuthClient(revokeReqDTO);
+    }
 
+    private OAuthRevocationRequestDTO getoAuthRevocationRequestDTO(OAuthRevocationRequestDTO oauthRequest) {
```

Comment:
Looks like this method is redundant. It takes **OAuthRevocationRequestDTO** object as parameter, build another **OAuthRevocationRequestDTO** object with details of the sent parameter. Can we remove this method and directly call getOAuth2Service().revokeTokenByOAuthClient(oauthRequest) in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/667/files#diff-c9fcb5ce9b0a94b5aa9ac6891ba5cba9R337

Reply:
Yes. Fixed